src/models/application/preferences.rs

```diff
@@ -37,6 +46,10 @@ pub struct Preferences {
     data: Option<Yaml>,
     keymap: KeyMap,
     theme: Option<String>,
+    // We use a refcell here to be able to mutate (insert into) the cache and
+    // still have all the data access methods take &self instead of &mut self.
+    // See https://doc.rust-lang.org/std/cell/index.html#implementation-details-of-logically-immutable-methods
+    editorconfig_cache: RefCell<HashMap<String, OrderMap<String, String>>>,
```

Comment:
I'd rather see us eager load the editorconfig file in the constructor than use interior mutability to lazily load it afterwards. Given that, this would wind up being `Option<HashMap<String, OrderMap<String, String>>`, and would work like `data`, where it's only set via `Preferences::load`.

Reply:
I also though about eagerly-loading .editorconfig before (since it is ), but then how should we deal with complex usage scenarios using "nested" .editorconfig files?

Consider the following directory structure:
```
project
├── .editorconfig [with root=true]
├── a
│   ├── .editorconfig
│   ├── 1.rs
│   └── c
│       ├── .editorconfig
│       └── 3.rs
└── b
    └── 2.rs
```

In this scenario, there may be rules which apply to `a/1.rs` due to being specified in `a/.editorconfig`, but are not applicable for `a/c/3.rs` or `b/2.rs`.

By lazy-loading the specific options for each file upon querying, all directories until hitting a root-.editorconfig can be easily crawled and parsed.

This could probably be implement without too much fuzz by pre-loading all .editorconfig and then just path-matching, but the editorconfig-library used here cannot handle something like that currently - I will have a look at it over the next few days.

Reply:
As a first attempt, I'd limit the scope of this feature to a `.editorconfig` file in the workspace directory and ignore this nested configuration altogether. :slightly_smiling_face: